src/WebJobs.Extensions.CosmosDB/Trigger/CosmosDBTriggerObserver.cs

```diff
@@ -38,9 +40,31 @@ public Task OpenAsync(ChangeFeedObserverContext context)
             return Task.CompletedTask;
         }
 
-        public Task ProcessChangesAsync(ChangeFeedObserverContext context, IReadOnlyList<Document> docs)
+        public async Task ProcessChangesAsync(ChangeFeedObserverContext context, IReadOnlyList<Document> docs)
         {
-            return this.executor.TryExecuteAsync(new TriggeredFunctionData() { TriggerValue = docs }, CancellationToken.None);
+            int retries = 0;
+
+            while (true)
+            {
+                var result = await this.executor.TryExecuteAsync(new TriggeredFunctionData() { TriggerValue = docs }, CancellationToken.None);
+
+                if (result.Succeeded)
+                {
+                    // notification processed successfully
+                    return;
+                }
+
+                // note: -1 means "infinite retries", so ignoring the count of retries in that case
+                if (retryCount != -1 && retries >= retryCount)
+                {
+                    // Function invocation failed, but no retries are left
+                    return;
+                }
+
+                // Function invocation failed, but retries are left, or infinite retries was specified
+                retries++;
+                continue;
+            }
```

Comment:
It'd be good to get some trace logging in here. I can plumb that through.

Reply:
Thanks!